MT node cache is not writing to DB in some cases

Running `make run && sleep 3 && make init-network && make deploy-uniswap` from develop:
```
2022-04-18T17:25:59+02:00	ERROR	scripts/scripts.go:35	tx mined[FAILED]: 0x470381f005b39e1b32043694b26d09dfb6637a1a3adecf946c9c24e1ed3d9463
2022-04-18T17:25:59+02:00	DEBUG	uniswap/main.go:138	Add Liquidity to Pair A <-> wEth tx: 0x470381f005b39e1b32043694b26d09dfb6637a1a3adecf946c9c24e1ed3d9463

2022-04-18T17:25:59+02:00	DEBUG	uniswap/main.go:302	Adding liquidity(10000000000000000) for tokens A: 0x4B53839f1FE653fF86c077F9c618191803A3e9A9, B:0x3A07588DefB088956a2e6dD15C33d63F2E0A2c55, Recipient: 0xC949254d682D8c9ad5682521675b8F43b102aec4
2022-04-18T17:25:59+02:00	INFO	scripts/scripts.go:17	waiting tx to be mined: 0xf51838bcaa81e362b5aa84ccd239b53c13555089065f6b19dbdd8202590233cd
2022-04-18T17:26:04+02:00	ERROR	scripts/scripts.go:30	Failed to get tx receipt: Post "http://localhost:8123/": dial tcp [::1]:8123: connect: connection refused%!(EXTRA string=
/home/fgimenez/workspace/polygon-hermez/hermez-core/log/log.go:106 github.com/hermeznetwork/hermez-core/log.appendStackTraceMaybeArgs()
/home/fgimenez/workspace/polygon-hermez/hermez-core/log/log.go:164 github.com/hermeznetwork/hermez-core/log.Errorf()
/home/fgimenez/workspace/polygon-hermez/hermez-core/scripts/scripts.go:30 github.com/hermeznetwork/hermez-core/scripts.WaitTxToBeMined()
/home/fgimenez/workspace/polygon-hermez/hermez-core/scripts/uniswap/main.go:305 main.addLiquidity()
/home/fgimenez/workspace/polygon-hermez/hermez-core/scripts/uniswap/main.go:141 main.main()
/usr/local/go/src/runtime/proc.go:255 runtime.main()
)
2022-04-18T17:26:04+02:00	FATAL	uniswap/main.go:317	Post "http://localhost:8123/": dial tcp [::1]:8123: connect: connection refused
/home/fgimenez/workspace/polygon-hermez/hermez-core/log/log.go:106 github.com/hermeznetwork/hermez-core/log.appendStackTraceMaybeArgs()
/home/fgimenez/workspace/polygon-hermez/hermez-core/log/log.go:137 github.com/hermeznetwork/hermez-core/log.Fatal()
/home/fgimenez/workspace/polygon-hermez/hermez-core/scripts/uniswap/main.go:317 main.chkErr()
/home/fgimenez/workspace/polygon-hermez/hermez-core/scripts/uniswap/main.go:306 main.addLiquidity()
/home/fgimenez/workspace/polygon-hermez/hermez-core/scripts/uniswap/main.go:141 main.main()
/usr/local/go/src/runtime/proc.go:255 runtime.main()

exit status 1
make: *** [Makefile:140: deploy-uniswap] Error 1
```
and in the core container logs:
```
2022-04-18T15:26:04Z	INFO	state/batchprocessor.go:112	Successfully processed transaction 0xf51838bcaa81e362b5aa84ccd239b53c13555089065f6b19dbdd8202590233cd
2022-04-18T15:26:04Z	FATAL	synchronizer/synchronizer.go:261	error processing batch. BatchNumber: 19. Error: not found
/src/log/log.go:106 github.com/hermeznetwork/hermez-core/log.appendStackTraceMaybeArgs()
/src/log/log.go:137 github.com/hermeznetwork/hermez-core/log.Fatal()
/src/synchronizer/synchronizer.go:261 github.com/hermeznetwork/hermez-core/synchronizer.(*ClientSynchronizer).processBlockRange()
/src/synchronizer/synchronizer.go:177 github.com/hermeznetwork/hermez-core/synchronizer.(*ClientSynchronizer).syncBlocks()
/src/synchronizer/synchronizer.go:111 github.com/hermeznetwork/hermez-core/synchronizer.(*ClientSynchronizer).Sync.func1()
```
The error happens after a failed tx, in that case the nodes in the cache are not being written and maybe some parts of the code expects them to be present.

--- a/state/tree/cache.go
+++ b/state/tree/cache.go
@@ -36,7 +36,7 @@ type nodeCache struct {
 }
 
 const (
-	maxMTNodeCacheEntries = 256
+	maxMTNodeCacheEntries = 65536
 )
 
 var (

